File: Ejercicio03/MainActivity.java
package com.cesar.dado;

import android.os.Bundle;
import android.os.Handler;
import android.widget.Button;
import android.widget.ImageView;
import androidx.appcompat.app.AppCompatActivity;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

    ImageView imgDado;
    Button btnLanzar;

    int[] dados = {
            R.drawable.dado1,
            R.drawable.dado2,
            R.drawable.dado3,
            R.drawable.dado4,
            R.drawable.dado5,
            R.drawable.dado6
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        imgDado = findViewById(R.id.imgDado);
        btnLanzar = findViewById(R.id.btnLanzar);

        btnLanzar.setOnClickListener(v -> lanzarDado());
    }

    private void lanzarDado() {
        Random random = new Random();
        Handler handler = new Handler();

        for (int i = 0; i < 10; i++) {
            int delay = i * 100;

            handler.postDelayed(() -> {
                int temp = random.nextInt(6);
                imgDado.setImageResource(dados[temp]);
            }, delay);
        }

        handler.postDelayed(() -> {
            int resultado = random.nextInt(6);
            imgDado.setImageResource(dados[resultado]);
        }, 1000);
    }
}

File: Ejercicio02/MainActivity.kt
package com.cesar.pizza

import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val gridView = findViewById<GridView>(R.id.gridView)

        val nombres = arrayOf(
            "Americana",
            "Napolitana",
            "Pepperoni",
            "Queso",
            "Romana",
            "Siciliana"
        )

        val imagenes = arrayOf(
            R.drawable.americana,
            R.drawable.napolitana,
            R.drawable.peperoni,
            R.drawable.queso,
            R.drawable.romana,
            R.drawable.siciliana
        )

        val adapter = PizzaAdapter(this, nombres, imagenes)
        gridView.adapter = adapter

        gridView.setOnItemClickListener { _, _, position, _ ->
            Toast.makeText(this, "Pizza: ${nombres[position]}", Toast.LENGTH_SHORT).show()
        }
    }
}